### PR TITLE
Update lint-staged configuration in root & starter + remove lint-staged from showcase

### PR DESCRIPTION
**Pull request by SuperITMan**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [X] The commit message follows our guidelines: https://github.com/NationalBankBelgium/stark/blob/master/CONTRIBUTING.md#-commit-message-guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[X] Feature
[ ] Code style update (formatting, local variables)
[X] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

When we run `npm install && npm run install:all`, the husky hooks are not well configured because the "starter" is installed after the root. 
The result is when we do a commit, only the git hook set in the "starter" is triggered instead of the one configured in the root folder.

Also, the showcase contains the dependency "lint-staged" even if it is not needed neither used.

The husky config is not complete in the starter which means that no hook is actually triggered.

Issue Number: N/A

## What is the new behavior?

Updated lint-staged configuration in the root and in the starter, removing the useless "git add ." which is done automatically. 

When we run `npm run install:starter`, the "husky" dependency is not installed thanks to the variable "SKIP_HUSKY_INSTALL". 

Removed the useless "lint-staged" dependency from the "showcase".

## Does this PR introduce a breaking change?
```
[ ] Yes
[X] No
```

## Other information

### CODE REVIEW

File: starter/package.json [hunk 1]
```diff
@@ -62,7 +62,6 @@
     "postinstall": "npm run webdriver:update",
     "postversion": "git push && git push --tags",
     "preclean:install": "npm run clean",
-    "precommit": "lint-staged",
```
Comment (christophercr) [inline starter/package.json]: Why is this removed in the starter if `lint-staged` is still in the `devDependencies` list?

Comment (SuperITMan) [reply starter/package.json]: Because this script is useless.
It is not present in our main package.json neither

Comment (christophercr) [reply starter/package.json]: Ooops... I missed that, sorry :p
